Just examining and understanding the data

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("all_authors.csv")

First we check the average number of authors per publisher

In [4]:
# Group by publisher and count unique authors
authors_per_publisher = df.groupby("Publisher")["Author"].nunique().reset_index(name="Unique_Authors")
authors_per_publisher


Publisher  Unique_Authors
0                2Leaf Press              32
1              39 West Press              21
2                    47North              29
3        5 Prince Publishing              30
4               A BEKA Books              29
..                       ...             ...
607     bonbon Kids Magazine              25
608              eBook Voice              26
609        iLAB America Inc.              13
610        plusDA Publishers              12
611  publisher.amplecare.com               1

[612 rows x 2 columns]

Here we find that the average number of authors is 29.31

In [5]:
# Calculate the average number of authors across all publishers
average_authors = authors_per_publisher["Unique_Authors"].mean()
print(f"Average number of authors per publisher: {average_authors:.2f}")


Average number of authors per publisher: 29.31


Now we want to check if any publishers are missing from the list so we compare to the publishers.csv file

In [6]:
all_publishers_df = pd.read_csv("publishers.csv")


In [7]:
# Get unique publishers from each
publishers_in_books = set(df["Publisher"].dropna().unique())
publishers_in_list = set(all_publishers_df["Publisher"].dropna().unique())


Here we determine that 13 publishers are missing, meaning that we were not able to gather any data on these 13.

In [8]:
# Publishers in the full list but missing from the books dataset
missing_publishers = publishers_in_list - publishers_in_books

# Show results
print(f"Missing publishers count: {len(missing_publishers)}")
missing_publishers = sorted(missing_publishers)
missing_publishers[:10]  # Show first 10 as a sample


Missing publishers count: 13


['Aardvark to Zucchini Press',
 'Aojirouma Publishing',
 'Books Launch Vast New Possibilities (BLVNP) Incorporated',
 'DYSTENIUM LLC - Publishing for The Third Millennium™',
 'Feenixx Publishing, Inc.',
 'FutureClaw Magazine',
 'KugaliInk',
 'Mediaplanet Inc.',
 'ModoComm Communications',
 'Norman Maine Publishing']

That leaves us with an average of 29.31 authors for 612 publishers